### PR TITLE
fix: unwanted behaviour when the menu is not displayed

### PR DESCRIPTION
**Pull request by AnthonyDS-symphony**
For the need of [this CES ticket](https://perzoinc.atlassian.net/browse/CES-5147), I need to replace our DropDownMenu with the one from the UI Toolkit.

When implementing it, I found an issue because event listener `click` triggered the `mouseEventHandler` event for each click every where on the page **even if the menu is not open**.

So, when you click on a button that should open the menu, the `close()` is automatically triggered in the same time: so we can't open the menu.

I've just added a watch on `show` property in order to skip event listeners.
In short:
- `show=false`: we don't need to listen events
- `show=true`: we need to listen events

### CODE REVIEW

**Review by liuting1014 — APPROVED** [review]

**Comment by liuting1014** [comment]
It would be nice to add some unit tests :)

**Review by AliciaSymphony — APPROVED** [review]
Great improvement, thank you for your contribution!!

**Review by symphony-jean-michael — APPROVED** [review]
LGTM. Thank you for the fix @AnthonyDS-symphony ! 😃
